tests/spec/core/sections-spec.js

```diff
@@ -0,0 +1,87 @@
+"use strict";
+
+import { flushIframes, makeRSDoc, makeStandardOps } from "../SpecHelper.js";
+
+describe("Core — sections", () => {
+  afterAll(flushIframes);
+
+  it("wraps headings in sections elements and assigns an id", async () => {
+    const body = `
+      <h2 id="h2">Section 2</h2>
+      <h3 id="h3">Section 3</h3>
+      <h4 id="h4">Section 4</h4>
+      <h5 id="h5">Section 5</h5>
+      <h6 id="h6">Section 6</h6>
+    `;
+    const ops = makeStandardOps(null, body);
+    const doc = await makeRSDoc(ops);
+    for (let i = 2; i <= 6; i++) {
+      const h = doc.getElementById(`h${i}`);
+      const section = h.parentElement;
+      expect(section.localName).toBe("section");
+      expect(section.id).toBe(`section-${i}`);
+    }
+  });
+
+  it("it handles when some headings are wrapped", async () => {
+    const body = `
+      <h2 id="h2">Section 2</h2>
+      <section>
+        <h3 id="h3">Section 3</h3>
+        <section>
+          <h4 id="h4">Section 4</h4>
+          <h5 id="h5">Section 5</h5>
+          <section>
+            <h6 id="h6">Section 6</h6>
+          </section>
+        </section>
+      </section>
+    `;
+    const ops = makeStandardOps(null, body);
+    const doc = await makeRSDoc(ops);
+    for (let i = 2; i <= 6; i++) {
+      const h = doc.getElementById(`h${i}`);
+      const section = h.parentElement;
+      expect(section.localName).toBe("section");
+      expect(section.id).toBe(`section-${i}`);
+    }
+  });
+
+  it("retains different nodes when wrapping", async () => {
+    const body = `
+      <h2 id="h2">Section 2</h2>
+        <p>paragraph 2</p><!--comment 2-->
```

Comment:
Is this indent on `<p>` intentional/meaningful?

Reply:
Nope. Will fix.